students/carlos_novoa/lesson09/mailroom.py

```diff
@@ -0,0 +1,337 @@
+             '               Sincerely,\n'
+             '                  -The Team\n\n')
+        print(m.format(name, donations))
+
+    def generate_letters(self):
+        cwd = os.getcwd()
+        date = now.strftime('%Y-%m-%d')
+        path = cwd + '/letters/'
+        ext = '.txt'
+        for key, val in self.donors.items():
+            total_donations = sum(val)
+            n = key.split(' ')
+            file_path = "{}{}_{}_{}{}".format(path, date, n[0], n[1], ext)
+            donations = Helpers.format_currency_str(total_donations)
+            with open(file_path, 'w') as letter:
+                text = ('\n\nDear {} {},\n\n'
+                        '        Thank you for your very kind '
+                        'donation of {}.\n\n'
+                        '        It will be put to very good use.\n\n'
+                        '               Sincerely,\n'
+                        '                  -The Team\n\n')
+                body = text.format(n[0], n[1], donations)
+                letter.write(body)
+        print('\n\n========== Letters Created ==========\n\n')
+
+    def create_report(self):
+        sorted_donors = self.sort_donors()
+        rows = self.get_donor_summary(sorted_donors)
+        self.print_report(rows)
+
+    @staticmethod
+    def print_report(rows):
+        # table heading
+        h = ['Donor Name', 'Total Given', 'Num Gifts', 'Average Gift']
+        hs = ' | '
+        hf = '{0:<25}{1}{2}{3}{4}{5}{6}'
+        print(hf.format(h[0], hs, h[1], hs, h[2], hs, h[3]))
+        # table rows
+        for r in rows:
+            name = "{}".format(r[0])
+            f0 = '{0:<' + str(max(len(name), 25)) + '}'
+            f2 = '{2:>' + str(max(len(r[1]), len(h[1]))) + '}'
+            f4 = '{4:>' + str(max(len(r[2]), len(h[2]))) + '}'
+            f6 = '{6:>' + str(max(len(r[3]), len(h[3]))) + '}'
+            rf = f0 + '{1}' + f2 + '{3}' + f4 + '{5}' + f6
+            args = [name, '  $', r[1], ' | ', r[2], '  $', r[3]]
+            print(rf.format(*args))
+
+    def donor_summary(self, name):
+        summary = ('\n\n===========================\n'
+                   'Donor: {}\n'
+                   'Total Donations: {}.\n'
+                   'Donation Count: {}\n'
+                   'Donation Average: {}\n'
+                   '===========================\n\n')
+        total = self.donations_total(name)
+        count = self.donations_count(name)
+        average = self.donations_average(name)
+        print(summary.format(name, total, count, average))
+
+    def donations_total(self, name=None):
+        return Helpers.format_currency_str(sum(self.donors[name]))
+
+    def donations_count(self, name=None):
+        return len(self.donors[name])
+
+    def donations_average(self, name=None):
+        total = sum(self.donors[name])
+        count = len(self.donors[name])
+        return Helpers.format_currency_str(total / count)
+
+
+class Donor(Donors):
```

Comment:
Subclassing should almost always be used for 'is a' relationships, but 'a donor is a donors' does not make sense so this is a good indication that inheritance isn't the right relationship to establish between these classes. 

I would recommend having a Donor class that encapsulates all the logic and data related to a single donor, and a Donors class for logic and data related to the collection of donors as a whole, so you're on the right track, but you shouldn't need Donor to inherit all the behavior and attributes from Donors